pandas/core/apply.py

```diff
@@ -44,7 +49,38 @@ def frame_apply(
 
 
 class FrameApply:
-    def __init__(self, obj, func, raw, result_type, ignore_failures, args, kwds):
+
+    # ---------------------------------------------------------------
+    # Abstract Methods
+
+    @property
+    def axis(self) -> int:
+        raise AbstractMethodError(self)
+
+    @property
+    def result_index(self) -> "Index":
+        raise AbstractMethodError(self)
+
+    @property
+    def result_columns(self) -> "Index":
+        raise AbstractMethodError(self)
+
+    @property
+    def series_generator(self) -> Iterator["Series"]:
+        raise AbstractMethodError(self)
+
+    # ---------------------------------------------------------------
+
+    def __init__(
+        self,
+        obj: "DataFrame",
```

Comment:
should we consider making FrameApply generic if we want subclassed DataFrames to return same type?

Reply:
im not sure i understand the question.  FrameApply is the base class for ColumnFrameApply and RowFrameApply

Reply:
I've not looked in detail but may not be relevant here.

Not sure how geopandas (or other downstream packages work), Just keeping in mind that, say GeoDataFrame, may return a GeoDataFrame with certain operations.

did start to type geopandas so I had a better idea, see https://github.com/geopandas/geopandas/compare/master...simonjayhawkins:typing. 

Note to self: Should try and get back to this.